servers/zms/src/main/java/com/yahoo/athenz/zms/store/ObjectStoreConnection.java

```diff
@@ -146,4 +146,6 @@
     boolean updateRoleMemberReviewNotificationTimestamp(String server, long timestamp);
 
     DomainRoleMembers listOverdueReviewRoleMembers(String domainName);
+
+    Map<String, Integer> getDomainTemplates(String domainName);
```

Comment:
Not sure why we're returning Map<String, Integer> here. if we add additional attributes, then this is not going to work thus is going to be changed again.

Looking at the server side, it looks like we're taking this Map<String, Integer> and then we generate List<TemplateMetaData> by filling the rest of the fields in the template meta objects. So why not just return here. the api should be:

List<TemplateMetaData> getDomainTemplates(String domainName);

then you just need fill in the other fields before returning to the client. in the future, if we need to add additional attributes, it would just require updating the TemplateMetaData object and not the api itself

Reply:
replaced Map with List<TemplateMetaData> object